### DISCUSSION TITLE
Reset controlled component

### DISCUSSION THREAD

**Discussion by dan-cooke** [airjp73/rvf#144] [Q&A] [answered]
I'm having difficulties resetting my material-ui components.

With a vanilla HTML form element, its simple. 

But with my custom material-ui component, it seems nearly impossible.

```.tsx
export const ValidatedSelect: React.FC<ValidatedSelectProps> = ({
  name,
  children,
  ...selectProps
}) => {
  const { error, getInputProps, validate, defaultValue } = useField(name);

  const [value, setValue] = useState<any>();

  useEffect(() => {
    validate();
  }, [value]);

  const inputProps = getInputProps(selectProps);

  return (
    <Select
      {...inputProps}
      onChange={(e) => {
        setValue(e.target.value);
        inputProps?.onChange?.(e);
      }}
      error={!!error}
      helperText={error}
      autoComplete={name}
    >
      {children}
    </Select>
  );
};

```

Calling `reset()` from the form context does nothing in this case, and same for all my other material ui components.

Any ideas how I can hack around this, I definetly won't be using Material on my next project after this issue.

**Comment by airjp73** [comment, marked as answer]
You'll want to use [`useControlField`](https://www.remix-validated-form.io/reference/use-control-field) for that.

**Reply by dan-cooke** [reply]
Duh. Thank you! I saw it in the docs earlier but I must have glanced over it
